akka-docs/src/main/paradox/persistence.md

```diff
@@ -30,12 +30,12 @@ to see what this looks like in practice.
 
 ## Introduction
 
-Akka persistence enables stateful actors to persist their internal state so that it can be recovered when an actor
-is started, restarted after a JVM crash or by a supervisor, or migrated in a cluster. The key concept behind Akka
-persistence is that only changes to an actor's internal state are persisted but never its current state directly
-(except for optional snapshots). These changes are only ever appended to storage, nothing is ever mutated, which
-allows for very high transaction rates and efficient replication. Stateful actors are recovered by replaying stored
-changes to these actors from which they can rebuild internal state. This can be either the full history of changes
+Akka persistence enables stateful actors to persist their state so that it can be recovered when an actor
+is either restarted, such as after a JVM crash, by a supervisor or a manual stop-start, or migrated within a cluster. The key concept behind Akka
+persistence is that only the _events_ received by the actor are persisted, not the actual state of the actor
```

Comment:
The actor receives commands (`ChangeTheName(newName)`) which are then validated and events (`NameChanged(newName)`) are created and those are persisted, so maybe "_events_ representing changes that happened to the actor state" or something like that?

Reply:
Do all events represent changes in state?

What is it you're trying to achieve with your wording?

Reply:
@dwijnand I think the wording is a bit dense. How about:


```
Akka persistence enables stateful actors to persist their state so that it can be recovered when an actor is either restarted, such as after a JVM crash, by a supervisor or a manual stop-start, or migrated within a cluster. The key concept behind Akka persistence is that it's not the actor's state that is persisted but each change on the state. These state changes are called events. The events are persisted by appending to storage (no data in the storage is ever mutated) which allows for very high transaction rates and efficient replication. A stateful actor is recovered by replaying the stored events to the actor, allowing it to rebuild its state. This reconstruction operation is called rehydration. The rehydration can be either the full replay of events or starting from a snapshot (when available) dramatically reducing recovery times. 
```
I tried to focus on the concepts (e.g. removed the use of `receive`) and introduced some common language gradually (events, rehydration,...).

Finally, I don't know what the last sentence means:
```
Akka persistence also provides point-to-point communication with at-least-once message delivery semantics.
```

Reply:
Now you got me thinking it shouldn't say "enables stateful actors to persist their state" either...

Reply:
>  "enables stateful actors to persist their state" 

Hmmm, good point. I guess we could say:

```
Akka persistence adds durability to stateful actors so that the state can 
be recovered when an actor is either restarted, such as after a JVM crash, by a supervisor or a manual stop-start, or migrated within a cluster.
```
WDYT?

Reply:
I think it's better.  But perhaps it's worth add a short definition of "durability".

Also, I've been thinking that it might be better (in your previous update) to lead with technical terms first and backfill the meaning.  So instead of

> A stateful actor is recovered by replaying the stored events to the actor, allowing it to rebuild its state. This reconstruction operation is called rehydration.

something like

> A stateful actor is recovered (by/with/via) rehydration, (that is?) by replaying the stored events to the actor allowing it to rebuild its state.

I'm not sure if it's a good idea.

Reply:
SGTM

Reply:
> What is it you're trying to achieve with your wording?

Get the introduction to use event sourcing terminology but still explain in a clear way what it does (and not something it doesn't do).

I think I like @ignasi35 suggestion best so far.

> Finally, I don't know what the last sentence means:
>> Akka persistence also provides point-to-point communication 
>> with at-least-once message delivery semantics.

Akka Persistence contains the `AtLeastOnceDelivery` abstraction that does what it says on top of persistence.

Reply:
One thought though is that even if I have seen the term `rehydrating` in event sourcing contexts, the persistence APIs and existing docs use the terms `recover` the state by `replay` of the events, so maybe we should stick with those.

Reply:
I didn't specify which Monday.  ⌛️